java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -582,6 +917,8 @@ public void deleteRecursive(final String path) throws IOException {
                     break;
                 } else if (e.getServerCode().equals(MantaErrorCode.DIRECTORY_NOT_EMPTY_ERROR)) {
                     continue;
+                } else if (e.getServerCode().equals(MantaErrorCode.BUCKET_NOT_EMPTY_ERROR)) {
```

Comment:
I don't understand what it means to recursively delete a bucket.  Buckets can't have buckets inside them, right?

Reply:
It got left there when I was considering implementing `destroyBucket` as an API Call, it is an unnecessary check and will be removed.